virocon/distributions.py

```diff
@@ -68,18 +87,153 @@ def _get_param_values(self, given):
         return param_values
 
     def pdf(self, x, given):
+        """
+        Probability density function.
+
+        f_X(x) indicates the probability density at point x. 
+        It is the limit of the probability of the interval [x,x+Δ] divided by 
+        the length of the interval as the length of the interval goes to 0 
+        (x is a realization of a random variable X). 
+        
+            :math:`f_X(x) =  \lim_{x\\to 0}\\frac{P(x<X \leq x+ \\Delta)}{ \\Delta}`
+           
+        Parameters
+        ----------
+        x : array_like
+            Points at which the pdf is evaluated.
+            Shape:
+        
+        given : float or array_like
+           The conditioning value of the conditioning variable e.g. the
+           y in x|y.  
+           Shape:
+            
+        Returns
+        -------
+        ndarray
+            Probability densities at x.
+            Shape:
+        
+        
+        """
+        
         return self.distribution.pdf(x, **self._get_param_values(given))
     
     def cdf(self, x, given):
+        """
+        Cumulative distribution function.
+        
+        The cumulative distribution function of a continous random variable  X 
+        is the function given by:
+        
+            :math:`F_X(x) =  P(X\leq x)`    
+        
+        or:
+             
+            :math:`F_X(x) =  \int_{- \infty}^{x} f_X(t) dt`   
+     
+        Parameters
+        ----------
+        x : array_like
+            Points at which the cdf is evaluated.
+            Shape:
+        
+        given : float or array_like
+           The conditioning value of the conditioning variable e.g. the
+           y in x|y.  
+           Shape:
+   
+        Returns
+        -------
+        ndarray
+            Cumulative distribution function evaluated at x.
+            Shape:
+        
+        """
+       
         return self.distribution.cdf(x, **self._get_param_values(given))
     
     def icdf(self, prob, given):
+        """
+        Inverse cumulative distribution function.
+        
+        Calculates percent-point function. 
```

Comment:
Also known as quantile or percent-point function. 
With x a realization of this RV and given a realisation of the conditioning RV,
icdf(x, given) returns the inverse cumulative distribution function at x conditioned on given.